cryptoki/src/context/mod.rs

```diff
@@ -94,21 +94,37 @@ impl Pkcs11 {
         unsafe {
             let pkcs11_lib =
                 cryptoki_sys::Pkcs11::new(filename.as_ref()).map_err(Error::LibraryLoading)?;
-            let mut list = mem::MaybeUninit::uninit();
+            Self::_new(pkcs11_lib)
+        }
+    }
+
+    /// Instantiate a new context from current executable, the PKCS11 implementation is contained in the current executable
+    pub fn new_from_self() -> Result<Self> {
+        unsafe {
+            #[cfg(not(windows))]
+            let this_lib = libloading::os::unix::Library::this();
+            #[cfg(windows)]
+            let this_lib = libloading::os::windows::Library::this();
```

Comment:
Apparently the Windows variant returns a Result: https://docs.rs/libloading/latest/libloading/os/windows/struct.Library.html#method.this

```suggestion
            let this_lib = libloading::os::windows::Library::this()?;
```